### DISCUSSION TITLE
Chained actions

### DISCUSSION THREAD

**Discussion by forexknight** [filamentphp/filament#11906] [Help]
### Package

Table builder / Actions

### Package Version

v3.2

### How can we help you?

Hello,
I'm trying to figure out how can I achieve chained actions. By chained actions I mean that after edit action I want to trigger another action.

I have `UserResource` and within this resource I have `UserResource::table` method with such action which opens in modal:

```php 

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                ...
            ])
            ->actions([
               ...
                EditAction::make()
                    ->iconButton()
                    ->labeledFrom('sm')
                    ->after(function (User $record) {
                        ....
                    })
                    ->action(function (array $arguments) {
                        $this->replaceMountedAction('publish', $arguments);
                    }),

            ])
         ])
  }
```

```php 
// inside UserResource
public function publishAction(): Action
{
        return Action::make('publish')
            ->requiresConfirmation()
            ->action(function (array $arguments) {

            });
 }
```

I was refering to the [docs ](https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component#chaining-actions)and I also had publish method inside this resource but then I got exception that publish action is not defined in current context. It's actually logical, because in closure it is in different context. 

How can I achieve effect of chained actions?

**Comment by leandrocfe** [comment]
1 - Add the publish action in the ListPage or ManagePage if you are using simple resource

```php
public function publishAction(): Action
{
    return Action::make('Publish')
        ->requiresConfirmation()
        ->action(function () {
            //...
        });
}
```

2 - Mount the action
```php
Tables\Actions\Action::make('custom_edit')
    ->action(function ($livewire) {
        $livewire->mountAction('publish');
    })
```
